examples/assets/scripts/utils/cubemap-renderer.js

```diff
@@ -0,0 +1,106 @@
+//   and properties of the Camera are used to render cubemap (adjust near / far distance, clearing, layers and other properties)
+//   Note: the layers should contain all layers visible by cybemap camera.
+// - to use generated cube map, you can access it like this using script:
+//   material.cubeMap = entity.script.cubemapRenderer.cubeMap;
+
+var CubemapRenderer = pc.createScript('cubemapRenderer');
+
+CubemapRenderer.attributes.add('resolution', {
+    title: 'Resolution',
+    description: 'Resolution of one side of a cubemap. Use power of 2 resolution if you wish to use Mipmaps.',
+    type: 'number',
+    default: 64
+});
+
+CubemapRenderer.attributes.add('mipmaps', {
+    title: 'Mipmaps',
+    description: 'If set to true, mipmaps will be allocated and autogenerated.',
+    type: 'boolean',
+    default: false
+});
+
+CubemapRenderer.attributes.add('depth', {
+    title: 'Depth',
+    description: 'If set to true, depth buffer will be created.',
+    type: 'boolean',
+    default: true
+});
+
+// initialize code called once per entity
+CubemapRenderer.prototype.initialize = function () {
+
+    // this entity needs to have camera component as well
+    var camera = this.entity.camera;
+    if (!camera) {
+        console.error("CubemapRenderer component requires Camera component to be created on the Entity.");
+        return;
+    }
+
+    // disable camera component, as it's used only as a source of properties
+    camera.enabled = false;
+
+    // Create cubemap render target with specified resolution and mipmap generation
+    this.cubeMap = new pc.Texture(this.app.graphicsDevice, {
+        width: this.resolution,
+        height: this.resolution,
+        format: pc.PIXELFORMAT_R8_G8_B8_A8,
+        cubemap: true,
+        mipmaps: this.mipmaps,
+        autoMipmap: this.mipmaps,
+        minFilter: pc.FILTER_LINEAR_MIPMAP_LINEAR,
+        magFilter: pc.FILTER_LINEAR
+    });
+
+    // angles to render camera for all 6 faces
+    var cameraRotations = [
+        new pc.Quat().setFromEulerAngles(0, 90, 180),
+        new pc.Quat().setFromEulerAngles(0, -90, 180),
+        new pc.Quat().setFromEulerAngles(90, 0, 0),
+        new pc.Quat().setFromEulerAngles(-90, 0, 0),
+        new pc.Quat().setFromEulerAngles(0, 180, 180),
+        new pc.Quat().setFromEulerAngles(0, 0, 180)
+    ];
+
+    // set up rendering for all 6 faces
+    for (var i = 0; i < 6; i++) {
+
+        // render target, connected to cubemap texture face
+        var renderTarget = new pc.RenderTarget(this.app.graphicsDevice, this.cubeMap, {
+            depth: this.depth,
+            face: i
+        });
+
+        // create a child entity with the camera for this face
+        var e = new pc.Entity("CubeMapCamera_" + i);
+        e.addComponent('camera', {
+            aspectRatio: 1,
+            fov: 90,
+
+            // cubemap will render all layers as setup on Entity's camera
+            layers: camera.layers.slice(),
```

Comment:
You don't. 😆  It's redundant.

Reply:
See here for why: https://github.com/playcanvas/engine/blob/master/src/scene/camera.js#L223

Reply:
here's the reason: https://github.com/playcanvas/engine/blob/6d37be6b40660648a5cee2939fe66cc973204da2/src/framework/components/camera/component.js#L104-L107

Reply:
Haha, so I think those docs aren't quite right. At least, not according to the behavior of the code.

Reply:
It should say that `To update layers, don't get and edit the array. Always set the layers property to trigger an update.`

Reply:
Soooo, are you gonna remove the redundant `slice`? We can tighten up the docs for `pc.CameraComponent#layers` in a separate PR.

Reply:
I'm not sure the slice is redundant.
We have source camera with Layers, which is an array
and I'm assigning it to Layers of another camera on this line
without slice, both cameras will have a single instance of the array .. so if we change its one camera will change its content, it will change for the other camera as well
as this is code of setter of layer:

```
set layers(newValue) {
        this._camera.layers = newValue;
}
```

so I'd prefer to use slice to have two separate arrays.

but maybe a better solution would be to slice array inside set layers.

Reply:
No, scroll up to my earlier comments on this where I link to the line of code that show `Camera` doing its own slice of the supplied array.

Reply:
nice one, you looked one layer deeper than me. Removed the slice now.